### PR TITLE
[Component] Tab#347: SwiftUI Tab

### PR DESCRIPTION
**Pull request by michael-zimmermann**
Tab Component in SwiftUI

### Pull requests template

This pull request has:
- [x] Code documentation on publics
- [x] Accessibility
	- [x] Dynamic sizes
	- [x] Identifiers
- [x] Tests
- [x] Demo update
- [x] Wiki

### CODE REVIEW

File: core/Sources/Components/Tab/View/UIKit/TabItemUIView.swift [hunk 1]
```diff
@@ -47,10 +47,14 @@ public final class TabItemUIView: UIControl {
         return border
     }()
 
-    // An internal property to determine if the segment width should be aligned to the
+    // An internal property to determin if the segment width should be aligned to the
```
Comment (LouisBorleeAdevinta) [inline core/Sources/Components/Tab/View/UIKit/TabItemUIView.swift:50]: isn't it "to determine"?

File: core/Sources/Components/Tab/ViewModel/TabViewModel.swift [hunk 2]
```diff
@@ -8,31 +8,54 @@
...
             self.tabsAttributes = self.useCase.execute(theme: theme, isEnabled: self.isEnabled)
         }
     }
 
-    var isEnabled: Bool = true {
-        didSet {
-            self.tabsAttributes = self.useCase.execute(theme: theme, isEnabled: self.isEnabled)
+    var isEnabled: Bool {
+        get {
+            return self.disabledTabs.reduce(true) { return $0 && !$1 }
```
Comment (LouisBorleeAdevinta) [inline core/Sources/Components/Tab/ViewModel/TabViewModel.swift:25]: maybe a short comment on this

File: core/Sources/Components/Tab/Properties/TabItemContent.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    /// An optional attributed title. If a title is set, this will have preference over an attributed title
+    public var attributedTitle: AttributedString?
+
+    /// An optional badge
+    public var badge: BadgeView?
+
+    /// Return true if either title or attributed title have a value
+    public var hasTitle: Bool {
+        return self.title != nil || self.attributedTitle != nil
```
[the remark this answers is not in the document]
Comment (michael-zimmermann) [reply core/Sources/Components/Tab/Properties/TabItemContent.swift:38]: If the text is empty, then the label is hidden in the stack view.

File: core/Sources/Components/Tab/Properties/TabUIItemContent.swift [hunk 4]
```diff
@@ -11,9 +11,13 @@ import UIKit
...
 /// - text: The text of the tab item.
-public struct TabUIItemContent: Equatable, Updateable {
-    public internal (set) var icon: UIImage?
-    public internal (set) var title: String?
+public struct TabUIItemContent: TitleContaining, Equatable, Updateable {
+    public var icon: UIImage?
+    public var title: String?
+
+    public var hasTitle: Bool {
+        return title != nil
```
[the remark this answers is not in the document]
Comment (michael-zimmermann) [reply core/Sources/Components/Tab/Properties/TabUIItemContent.swift:19]: Same as above. The label is hidden in the stack view, when there is no title.

File: core/Sources/Components/Tab/View/UIKit/TabItemUIView.swift (hunk 1 above)
Comment (michael-zimmermann) [reply core/Sources/Components/Tab/View/UIKit/TabItemUIView.swift:50]: Yes, and I'd already corrected it somewhere. But with all the rebases I've been doing, it keeps getting broken.

File: core/Sources/Components/Tab/Properties/TabItemContent.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (LouisBorleeAdevinta) [reply core/Sources/Components/Tab/Properties/TabItemContent.swift:38]: I meant, what if the title (or attributedTitle) is not nil but is ""? hasTitle returns true but should it?

[the remark this answers is not in the document]
Comment (michael-zimmermann) [reply core/Sources/Components/Tab/Properties/TabItemContent.swift:38]: I think I'd leave that to the programmer to decide if an empty string should be added.

**Review by LouisBorleeAdevinta — APPROVED** [review]

**Pull request merged**